mlx/backend/metal/unary.cpp

```diff
@@ -0,0 +1,233 @@
+// Copyright © 2024 Apple Inc.
+
+#include <fmt/format.h>
+#include <cassert>
+
+#include "mlx/backend/common/compiled.h"
+#include "mlx/backend/metal/compiled_preamble.h"
+#include "mlx/backend/metal/device.h"
+#include "mlx/backend/metal/kernels/defines.h"
+#include "mlx/backend/metal/utils.h"
+#include "mlx/primitives.h"
+
+namespace mlx::core {
+
+constexpr std::string_view unary_kernels = R"(
+[[kernel]] void {0}_v(
+    device const {1}* in,
+    device {2}* out,
```

Comment:
I put the kernels here (binary kernels in the `binary.cpp`, and so on). Maybe better to put them in a different file in `kernels/` that gets included? Not sure if either of you have a preference there @angeloskath @jagrit06

Reply:
I kinda like it like this. If we do want to do this for more complicated kernels maybe we need a solution like the preamble but for unary, binary, ternary this is pretty great imho.

Reply:
I think it will likely be a combination of the two. Anything that needs to be formatted at runtime will likely be like this. And we can try to keep that bit to a minimum by having it be just the instantiations essentially.

The other stuff I will probably put in preambles. But I don't think it makes sense to do it as one giant preamble since that won't scale. So likely I will change the preamble / include stuff to be a little more modular